tuf/client/client.go

```diff
@@ -46,493 +42,197 @@ func (c *Client) Update() error {
 	err := c.update()
 	if err != nil {
 		logrus.Debug("Error occurred. Root will be downloaded and another update attempted")
-		if err := c.downloadRoot(); err != nil {
+		logrus.Debug("Resetting the TUF builder...")
+
+		repoSoFar := c.builder.GetRepo()
+		c.builder = c.builder.BootstrapNewBuilder()
```

Comment:
Ah ok, basically we wanted to throw out any existing information we have and restart building a `tuf.Repo` object all over again fresh, so we can't just accidentally overwrite some previous data and end up with a half new half old repo.